@here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts

```diff
@@ -1805,6 +1776,207 @@ export class OmvDecodedTileEmitter implements IOmvEmitter {
         }
     }
 
+    private computeUntiledLine(polyline: ILineGeometry) {
+        // Compute the world position of the untiled line and its distance to the origin of the
+        // line to properly join lines.
+        const untiledLine = ([] as unknown) as UntiledLine;
+        untiledLine.lineDist = 0;
+        if (polyline.untiledPositions) {
+            this.m_decodeInfo.targetProjection.projectPoint(polyline.untiledPositions![0], tmpV3r);
+            polyline.untiledPositions!.forEach(pos => {
+                // Calculate the distance to the next unnormalized point.
+                this.m_decodeInfo.targetProjection.projectPoint(pos, tmpV3);
+                untiledLine.lineDist += tmpV3.distanceTo(tmpV3r);
+                tmpV3r.copy(tmpV3);
+
+                // Pushed the normalized point for line matching.
+                this.m_decodeInfo.targetProjection.projectPoint(pos.normalized(), tmpV3);
+                untiledLine.push(tmpV3.x, tmpV3.y, tmpV3.z, untiledLine.lineDist);
+            });
+        }
+
+        return untiledLine;
+    }
+
+    private convertLines(
+        lines: number[][],
+        uvs: number[][],
+        offsets: number[][],
+        geometry: ILineGeometry[],
+        extents: number,
+        computeTexCoords?: TexCoordsFunction
+    ) {
+        for (const polyline of geometry) {
+            const line: number[] = [];
+            const lineUvs: number[] = [];
+            const lineOffsets: number[] = [];
+            const untiledLine = this.computeUntiledLine(polyline);
+
+            polyline.positions.forEach(pos =>
+                this.addPoint(
+                    pos,
+                    line,
+                    lineUvs,
+                    lineOffsets,
+                    untiledLine,
+                    extents,
+                    computeTexCoords
+                )
+            );
+            lines.push(line);
+            uvs.push(lineUvs);
+            offsets.push(lineOffsets);
+        }
+    }
+
+    private convertAndCutLines(
```

Comment:
To avoid code duplication, you can maybe combine this function with the previous one by passing an extra optional parameter to the function (i.e. ``clipLines: boolean = false``)